cmd/build_test.go

```diff
@@ -14,6 +14,48 @@ import (
 	. "knative.dev/kn-plugin-func/testing"
 )
 
+// TestBuild_ImageFlag ensures that the image flag is used when specified.
+func TestBuild_ImageFlag(t *testing.T) {
+	var (
+		args    = []string{"--image", "docker.io/tigerteam/foo"}
+		builder = mock.NewBuilder()
+	)
+
+	root, cleanup := Mktemp(t)
+	defer cleanup()
+
+	// Write a func.yaml config which does not specify an image
+	funcYaml := `name: foo
+runtime: go
+builder: pack
+created: 2022-01-01T00:00:00+00:00
+`
+	if err := ioutil.WriteFile("func.yaml", []byte(funcYaml), 0600); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Since `func.yaml` is tightly coupled with the core `fn` library, and this `cmd` library depends on that API, I think it may be more usefule to use that api here as well:
```
  if err := fn.New().Create(fn.Function{Runtime: "go", Root: root}); err != nil {
    t.Fatal(err)
  }
```
You are then more guranteed that this unit test remains forward-compatible with what constitutes a valid function for purposes of the test, even if that may entail changes to the initial state of `func.yaml`.